pkg/roachprod/roachprod.go

```diff
@@ -1643,6 +1643,19 @@ func Grow(ctx context.Context, l *logger.Logger, clusterName string, numNodes in
 	return SetupSSH(ctx, l, clusterName)
 }
 
+func Shrink(ctx context.Context, l *logger.Logger, clusterName string, numNodes int) error {
+	c, err := getClusterFromCache(l, clusterName)
+	if err != nil {
+		return err
+	}
+
+	err = cloud.ShrinkCluster(l, &c.Cluster, numNodes)
+	if err != nil {
+		return err
+	}
+	return SetupSSH(ctx, l, clusterName)
```

Comment:
I agree with you that this is not strictly necessary, it may actually be okay to let the IPs linger in the certs.
Mmm, I'll think about keeping it, but definitely will add a doc to this and maybe grow as well.